pkg/cortex/modules.go

```diff
@@ -781,16 +781,20 @@ func (t *Cortex) initResourceMonitor() (services.Service, error) {
 			containerLimits[resource.Type(res)] = float64(runtime.GOMAXPROCS(0))
 		case resource.Heap:
 			containerLimits[resource.Type(res)] = float64(debug.SetMemoryLimit(-1))
+		default:
+			level.Warn(util_log.Logger).Log("msg", fmt.Sprintf("skipping unsupported resource [%s]", res))
 		}
 	}
 
 	var err error
 	t.ResourceMonitor, err = resource.NewMonitor(containerLimits, prometheus.DefaultRegisterer)
-	if t.ResourceMonitor != nil {
-		util_log.WarnExperimentalUse("resource monitor")
+	if err != nil {
```

Comment:
If it is a real error we usually fail it instead of logging a warning.
What could the scenario where this returns an error?

Reply:
I see, I wasn't sure if the resource monitor was critical enough to fail the whole program. But it's also true that we fail on any validation faults for configs as well. I will revert this change.